src/hw_ostc_parser.c

```diff
@@ -1021,8 +1059,15 @@ hw_ostc_parser_samples_foreach (dc_parser_t *abstract, dc_sample_callback_t call
 				return DC_STATUS_DATAFORMAT;
 			}
 			unsigned int o2 = data[offset];
+			unsigned int diluent;
+			if (version == 0x24) {
+				// all manually added gas mixes on OSTC4 are OC gases
+				diluent = 0;
```

Comment:
Besides the fact that the check for version 0x24 is wrong (see the other comment), the OSTC4 doesn't seem to produce manual gas change events in CC mode. So unless I missed something, I don't think this change will have any effect in practice. Did you encounter a problem here? Note that if the OSTC4 firmware ever gains support for manual gas change events, this code will be wrong.

In the OSTC4 code, the only place where the manualGasSet event is set is in the setActualGas_ExtraGas() function, and that is only called from OnAction_DM_Mix():
https://code.heinrichsweikamp.com/public/ostc4/file/tip/Discovery/Src/data_central.c#l538

Reply:
> the OSTC4 doesn't seem to produce manual gas change events in CC mode.

I think the `/* only for OC */` comment on `OnAction_DM_Mix()` is a bit misleading - what it means is that the `Extra Gas` change always results in the new gas being used as OC / bailout - it is not possible to switch to an extra diluent during the dive:

https://code.heinrichsweikamp.com/public/ostc4/file/tip/Discovery/Src/tMenuEditGasOC.c#l348

And yes, if the ability to switch to extra diluent gases during the dive is ever added for OSTC4 then this will have to be reworked. But I am not concerned about this, as currently the log format does not have a way to indicate that an extra gas being switched to is a diluent, so a protocol change (likely to break backwards compatibility) will have to happen for this.

Reply:
There is indeed no explicit indication whether a manual gas is an OC mix or diluent. That info is obtained indirectly from the dive mode. The assumption here is that all manual gas mixes are OC gas mixes when in OC mode and diluents when in CC mode. But that's the code you removed.

Note that bailout to OC has it's own event, so there we do know we're switching to an OC gas mix.

Reply:
> The assumption here is that all manual gas mixes are OC gas mixes when in OC mode and diluents when in CC mode. 

I try to base my code on established facts and not on assumptions, whereever this is possible.
And for the OSTC4 we *know* that a switch to an 'Extra Gas' is always treated as a switch to an *OC* gas - which is what I am modelling in the code that I am proposing.
Maybe we should try to get certainty on what is happening in the 'Extra Gas' case for the non-OSTC4 dive computers using this log format as well, to be able to base all of this code on established facts?
I am not familiar with the code base for these (or indeed what models and potential firmware versions exist), and I don't have the hardware to test this.
Or we could ask the manufacturer.